src/librustc/mir/mod.rs

```diff
@@ -2438,6 +2438,42 @@ EnumTypeFoldableImpl! {
     }
 }
 
+/// A collection of projections into user types.
+///
+/// They are projections because a binding can occur a part of a
+/// parent pattern that has been ascribed a type.
```

Comment:
It would be nice to give some Rust code examples here. I guess something like

```rust
let ((x, _): (T1, _)): (T2, _) = ...
```

is what you have in mind?

Reply:
I cannot decide whether something like `let a @ Thing1(_, b @ Thing2(_, _): Inner): Outer` would be clearer or not.

Reply:
okay added example to docs.